contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,876 @@
+        uint64 maxChangeRate; // maximum daily increase of the validator commission
+    }
+
+    struct SlashRecord {
+        uint256 jailUntil;
+        uint256 slashAmount;
+        uint248 slashHeight;
+        SlashType slashType;
+    }
+
+    enum SlashType {
+        DoubleSign,
+        DownTime,
+        MaliciousVote
+    }
+
+    /*----------------- events -----------------*/
+    event ValidatorCreated(
+        address indexed consensusAddress, address indexed operatorAddress, address indexed poolModule, bytes voteAddress
+    );
+    event ConsensusAddressEdited(
+        address indexed operatorAddress, address indexed oldAddress, address indexed newAddress
+    );
+    event CommissionRateEdited(address indexed operatorAddress, uint64 commissionRate);
+    event DescriptionEdited(address indexed operatorAddress);
+    event VoteAddressEdited(address indexed operatorAddress, bytes newVoteAddress);
+    event Delegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Undelegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Redelegated(
+        address indexed srcValidator,
+        address indexed dstValidator,
+        address indexed delegator,
+        uint256 oldShares,
+        uint256 newShares,
+        uint256 bnbAmount
+    );
+    event RewardDistributed(address indexed operatorAddress, uint256 reward);
+    event RewardDistributeFailed(address indexed operatorAddress, bytes failReason);
+    event ValidatorSlashed(
+        address indexed operatorAddress,
+        uint256 jailUntil,
+        uint256 slashAmount,
+        uint248 slashHeight,
+        SlashType slashType
+    );
+    event ValidatorJailed(address indexed operatorAddress);
+    event ValidatorEmptyJailed(address indexed operatorAddress);
+    event ValidatorUnjailed(address indexed operatorAddress);
+    event Claimed(address indexed operatorAddress, address indexed delegator, uint256 bnbAmount);
+    event StakingPaused();
+    event StakingResumed();
+
+    /*----------------- modifiers -----------------*/
+    modifier onlyInitialized() {
+        require(_initialized != 0, "NOT_INITIALIZED");
+        _;
+    }
+
+    modifier validatorExist(address operatorAddress) {
+        require(_validators[operatorAddress].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        _;
+    }
+
+    modifier validatorNotJailed(address operatorAddress) {
+        require(!_validators[operatorAddress].jailed, "VALIDATOR_JAILED");
+        _;
+    }
+
+    modifier whenNotPaused() {
+        require(!_stakingPaused, "STAKE_STOPPED");
+        _;
+    }
+
+    receive() external payable { }
+
+    /*----------------- init -----------------*/
+    function initialize() public onlyCoinbase onlyZeroGasPrice {
+        require(_initialized == 0, "ALREADY_INITIALIZED");
+
+        transferGasLimit = INIT_TRANSFER_GAS_LIMIT;
```

Comment:
We dont need define `INIT_TRANSFER_GAS_LIMIT` then,  just copy the vaule here.

Reply:
Done

Reply:
It is not fixed yet.

Reply:
Done